### PR TITLE
Update/animal catalog

### PR DESCRIPTION
**Pull request by bkjwjason**
I've made the following changes:

- Created fxml files for Animal class to link Animal Class to GUI
- Refactored MainApp.java to pull data from the AnimalCatalog instead
- Created other files (AnimalCatalog.java, AnimalModelManager.java) that were required for migration.

That being said, there are a few future changes to be done:

- In AnimalCatalog.java, exceptions are caught and system prints an error message. However, we need the command output of the GUI to show the error message instead.
- Storage Class has to be refactored for persistent storage to work.
- Test cases for all new methods.

Let us slowly migrate over to Animal class and remove unnecessary files in the future!

### CODE REVIEW

**Review by Blizzeracz — APPROVED** [review]
Looks pretty neat and tidy. LGTM!

File: src/main/java/seedu/address/model/AnimalModelManager.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    @Override
+    public void setAnimal(Animal target, Animal editedAnimal) {
+        requireAllNonNull(target, editedAnimal);
+
+        animalCatalog.setAnimal(target, editedAnimal);
+    }
+
+    //=========== Filtered Person List Accessors =============================================================
```
Comment (wjayee) [inline src/main/java/seedu/address/model/AnimalModelManager.java:116]: Should it be person or animal?

File: src/main/java/seedu/address/model/AnimalModelManager.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof ModelManager)) {
```
Comment (wjayee) [inline src/main/java/seedu/address/model/AnimalModelManager.java:140]: Should this be AnimalModelManager instead of ModelManager?

**Review by wjayee — CHANGES_REQUESTED** [review]
LGTM! Just a few changes to the classes and javadocs from person to animal :)

File: src/main/java/seedu/address/model/animal/Species.java [hunk 3]
```diff
@@ -47,6 +47,8 @@ public boolean equals(Object o) {
         return speciesName.equals(that.speciesName);
     }
 
+
+
```
[the remark this answers is not in the document]
Comment (bkjwjason) [reply src/main/java/seedu/address/model/animal/Species.java:51]: Yup, fixed!

File: src/main/java/seedu/address/model/AnimalModelManager.java (hunk 2 above)
Comment (bkjwjason) [reply src/main/java/seedu/address/model/AnimalModelManager.java]: Yes, thanks for pointing it out!

File: src/main/java/seedu/address/model/AnimalModelManager.java (hunk 1 above)
Comment (bkjwjason) [reply src/main/java/seedu/address/model/AnimalModelManager.java]: Yes, thanks for pointing it out!

File: src/main/java/seedu/address/model/AnimalModelManager.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    @Override
+    public void setAnimalCatalogFilePath(Path addressBookFilePath) {
+        requireNonNull(addressBookFilePath);
+        userPrefs.setAddressBookFilePath(addressBookFilePath);
+    }
+
+    //=========== AddressBook ================================================================================
+
+    @Override
+    public void setAddressBook(ReadOnlyAddressBook addressBook) {
```
[the remark this answers is not in the document]
Comment (bkjwjason) [reply src/main/java/seedu/address/model/AnimalModelManager.java]: Yup, I agree as well

File: src/main/java/seedu/address/model/AnimalCatalog.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+     */
+    {
+        animals = new UniqueAnimalList();
+    }
+
+    public AnimalCatalog() {}
+
+    /**
+     * Creates an AnimalCatalog using the Animals in toBeCopied
+     * @param toBeCopied Animals to be copied into the AnimalCatalog
```
[the remark this answers is not in the document]
Comment (bkjwjason) [reply src/main/java/seedu/address/model/AnimalCatalog.java:37]: I agree with the change!

**Review by wjayee — APPROVED** [review]
LGTM, good job with the UI!

**Pull request merged**